src/java.base/share/classes/java/lang/template/ValidatingProcessor.java

```diff
@@ -0,0 +1,163 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang.template;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import jdk.internal.javac.PreviewFeature;
+
+/**
+ * This interface describes the methods provided by a generalized string template processor. The
+ * primary method {@link ValidatingProcessor#process(StringTemplate)} is used to validate
+ * and compose a result using a {@link StringTemplate StringTemplate's} fragments and values lists.
+ * For example:
+ * {@snippet :
+ * class MyProcessor implements ValidatingProcessor<String, IllegalArgumentException> {
+ *     @Override
+ *     public String process(StringTemplate st) throws IllegalArgumentException {
+ *          StringBuilder sb = new StringBuilder();
+ *          Iterator<String> fragmentsIter = st.fragments().iterator();
+ *
+ *          for (Object value : st.values()) {
+ *              sb.append(fragmentsIter.next());
+ *
+ *              if (value instanceof Boolean) {
+ *                  throw new IllegalArgumentException("I don't like Booleans");
+ *              }
+ *
+ *              sb.append(value);
+ *          }
+ *
+ *          sb.append(fragmentsIter.next());
+ *
+ *          return sb.toString();
+ *     }
+ * }
+ *
+ * MyProcessor myProcessor = new MyProcessor();
+ * try {
+ *     int x = 10;
+ *     int y = 20;
+ *     String result = myProcessor."\{x} + \{y} = \{x + y}";
+ *     ...
+ * } catch (IllegalArgumentException ex) {
+ *     ...
+ * }
+ * }
+ * Implementations of this interface may provide, but are not limited to, validating
+ * inputs, composing inputs into a result, and transforming an intermediate string
+ * result to a non-string value before delivering the final result.
+ * <p>
+ * The user has the option of validating inputs used in composition. For example an SQL
+ * processor could prevent injection vulnerabilities by sanitizing inputs or throwing an
+ * exception of type {@code E} if an SQL statement is a potential vulnerability.
+ * <p>
+ * Composing allows user control over how the result is assembled. Most often, a
+ * user will construct a new string from the template string, with placeholders
+ * replaced by stringified objects from the values list.
```

Comment:
stringified looks funny - but have no idea on how to replace it

Reply:
```
 * Composing allows user control over how the result is assembled. Most often, a
 * user will construct a new string from the template string, with placeholders
 * replaced by string representations of value list elements. These string
 * representations are created as is invoking {@link String#valueOf}.
```